### ISSUE TITLE
Consolidate `configure` scripts

### ISSUE THREAD

**Issue by jakirkham** [openucx/ucx#8642]
Currently there is a top-level `configure` script generated by `autoconf`. Typically this is what most builders would use by default, but UCX has a [`configure-release` that is in `contrib`]( https://github.com/openucx/ucx/blob/5e32bd070d25203be4a17860ba473d026334d544/contrib/configure-release# ), which is [recommended by the docs]( https://openucx.readthedocs.io/en/master/running.html#building ). Though there is also [a multithreaded release script]( https://github.com/openucx/ucx/blob/5e32bd070d25203be4a17860ba473d026334d544/contrib/configure-release-mt ) as well.

Would it be possible to consolidate this into a single `configure` that sets most things up for a performant release build? Perhaps there are still some `configure` scripts for development that need to hang around in `contrib`, but this would make it easier for others building and shipping UCX to users to get started.

cc @pentschev @leofang

**Comment by yosefe** [comment]
@jakirkham to be clear, are you suggesting that the default ./configure will behave the same as ./contrib/configure-release?

**Comment by jakirkham** [comment]
Yeah that's right. Would that be possible? Or are there other things to consider before approaching such a change?

**Comment by shamisp** [comment]
@jakirkham It is quite challenging to come up with such config. For example, the most optimal build config enables u-arch optimizations that are not portable across machines. The multi-threaded build enables flows that will not be as optimal for apps that don't care about multithreading, etc. Essentially, the options and configurations that are optimal for one user may not be optimal for another user.

**Comment by yosefe** [comment]
IMO we can make the default configure closer to configure-release, but it still has to be multi-threaded and without non portable uarch optimizations

**Comment by shamisp** [comment]
@jakirkham  Maybe you can share some details, why you believe this is an issue today. thanks
